File: src/go/day1/puzzle1.go
package main

import (
	"fmt"
	"slices"
)

func puzzle1() {
	fmt.Println(sumdiffs(a1, a2))
}

func sumdiffs(a1, a2 []int) int {
	// a1 = slices.Clone(a1)
	// a2 = slices.Clone(a2)
	slices.Sort(a1)
	slices.Sort(a2)

	sum := 0
	for i := 0; i < len(a1); i++ {
		sum += abs(a1[i] - a2[i])
	}
	return sum
}

func abs(x int) int {
	if x < 0 {
		return -x
	}
	return x
}

File: src/python/day1/puzzle1.py
def sumdiffs(a, b):
    a = sorted(a)
    b = sorted(b)
    sum = 0
    for i in range(len(a)):
        sum += abs(a[i]-b[i])
    return sum

if __name__ == "__main__":
    print(sumdiffs([1, 2], [2, 1]))